service/controller/chart/v1/resource/release/create.go

```diff
@@ -42,6 +43,15 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		// use the default values and prevent errors on nested values.
 		err = r.helmClient.InstallReleaseFromTarball(ctx, tarballPath, ns, helm.ReleaseName(releaseState.Name), helm.ValueOverrides(releaseState.ValuesYAML))
 		if err != nil {
+			releaseContent, err := r.helmClient.GetReleaseContent(ctx, releaseState.Name)
+			if releaseContent.Status == helmFailedStatus {
```

Comment:
Check whether this is about release or not first, so we can jump into to `statusv1` resource.